relay-event-schema/src/protocol/span.rs

```diff
@@ -200,11 +200,11 @@ pub struct SpanData {
     pub db_system: Annotated<Value>,
 
     /// The sentry environment.
-    #[metastructure(field = "environment")]
+    #[metastructure(field = "sentry.environment", legacy_alias = "environment")]
```

Comment:
The disadvantage of this rename is that the new key will now be serialized into the payload. But `span.data` is not (yet) forwarded to sentry directly on standalone spans (we only extract some of it into `sentry_tags`, but that happens in relay). Spans as part of transactions are materialized into nodestore, but they don't rely on these particular `span.data` items AFAIK.